packages/botonic-plugin-contentful/src/cms/visitors/message-visitors.ts

```diff
@@ -64,3 +74,73 @@ export async function reachableFrom(
   }
   return reachable
 }
+
+/**
+ * To obtain which contents have references (eg. buttons) to a given
+ * content
+ */
+export class MessageContentInverseTraverser {
+  private readonly referencesTo = new Map<string, Set<MessageContent>>()
+
+  constructor(
+    readonly cms: CMS,
+    readonly context: Context,
+    readonly depth = 1
+  ) {
+    if (depth > 1) {
+      throw new CmsException(
+        'Depth>1 not supported yet for MessageContentInverseTraverser'
+      )
+    }
+  }
+
+  isLoaded(): boolean {
+    return this.referencesTo.size > 0
+  }
+
+  async load(fromContents?: MessageContent[]) {
+    fromContents =
+      fromContents ||
+      (await allContents<MessageContent>(
+        this.cms,
+        this.context,
+        MESSAGE_CONTENT_TYPES
+      ))
+
+    for (const fromContent of fromContents) {
+      const reachable = await reachableFrom(
```

Comment:
can this be parallelized?

Reply:
good point! Added in posterior PR 917cf23